fe1-web/model/objects/RollCall.ts

```diff
@@ -1,20 +1,73 @@
-import { LaoEvent, LaoEventState, LaoEventType } from 'model/objects/LaoEvent';
+import { LaoEvent, LaoEventState, LaoEventType } from './LaoEvent';
+import { Hash } from './Hash';
+import { Timestamp } from './Timestamp';
 
-// Plain-old-data
 export interface RollCallState extends LaoEventState {
-  // to be continued
+  name: string;
+  location: string;
+  description: string;
+  creation: number;
+  ongoing: boolean;
 }
 
 export class RollCall implements LaoEvent {
-  // to be continued
+  public readonly id: Hash;
+
+  public readonly name: string;
+
+  public readonly location: string;
+
+  public readonly description: string;
+
+  public readonly creation: Timestamp;
+
+  public readonly start: Timestamp;
+
+  public readonly end: Timestamp;
+
+  public readonly ongoing: boolean;
+
+  /* Not yet implemented:
+   * This object should probably also keep a list of the time periods
+   * during which the roll call was open (or re-opened).
+   */
 
   constructor(obj: Partial<RollCall>) {
-    // to be continued
+    if (obj === undefined || obj === null) {
+      throw new Error('Error encountered while creating a RollCall object: '
+      + 'undefined/null parameters');
+    }
+
+    if (!obj.id) throw new Error("Undefined 'id' when creating 'RollCall'");
+    if (!obj.name) throw new Error("Undefined 'name' when creating 'RollCall'");
+    if (!obj.location) throw new Error("Undefined 'location' when creating 'RollCall'");
+    if (!obj.description) throw new Error("Undefined 'description' when creating 'RollCall'");
+    if (!obj.creation) throw new Error("Undefined 'creation' when creating 'RollCall'");
+    if (!obj.start) throw new Error("Undefined 'start' when creating 'RollCall'");
+    if (!obj.end) throw new Error("Undefined 'end' when creating 'RollCall'");
+    if (!obj.ongoing) throw new Error("Undefined 'ongoing' when creating 'RollCall'");
```

Comment:
Will this work since `ongoing` is a boolean?

Reply:
Of course not 😆  thanks for the catch